storybookjs/storybook
# Manager: Update `store.settings.lastTrackedStoryId`

Closes #23723





## What I did

In `/code/ui/manager/src/components/layout/mobile.ts` when navigating between SideBar (`handleSideBarClick()`) and Canvas (`handleCanvasClick()`), `api.retrieveSelection()` is used to get the current story id and if the returned id is falsy, the first story is selected as the current story.

`api.retrieveSelection()` returns `store.getState().settings.lastTrackedStoryId`, however, this `lastTrackedStoryId` is not set anywhere in the code, causing the returned value to always be `null` and thus when switching between SideBar and Canvas, always the first story is displayed.

When a story link is clicked on the SideBar, the `api.selectStory()` is called with the story id.

To fix the issue, I have updated the `code/lib/manager-api/src/modules/stories.ts` file and added the code to update the `store.settings.lastTrackedStoryId` with the current story id when `api.selectStory()` is called.

I have also added a test for this behavior to the `code/lib/manager-api/src/tests/stories.test.ts` file.

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

1. Run the default sandbox
2. Switch to the responsive/mobile mode in DevTools or browse on a mobile device
3. Select / Click on a story on the sidebar
4. Select / Click on the Canvas nav link (at the bottom)
5. The same story should still be showing



### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
